### DISCUSSION TITLE
cogs

### DISCUSSION THREAD

**Discussion by Saadelamali** [DisnakeDev/disnake#552] [Q&A]
### Summary

The slash commands in cogs dont work. It used to work fine 

### Reproduction Steps

just loading the cogs

### Expected Results

shows the slash commands in the (test_guilds)

### Actual Results

The commands work fine but slash commands dont even show up in the specific guild (test_guilds)

### Intents

members

### System Information

```markdown
- Python v3.10.2-final
- disnake v2.5.1-final
    - disnake pkg_resources: v2.5.1
- aiohttp v3.7.4.post0
- system info: Windows 10 10.0.19044
```

### Checklist

- [X] I have searched the open issues for duplicates.
- [X] I have shown the entire traceback, if possible.
- [X] I have removed my token from display, if visible.

**Comment by helish88** [comment]
ask for help here https://discord.gg/disnake

**Comment by Saadelamali** [comment]
so you're saying that's not a bug?

**Comment by helish88** [comment]
I'm sorry, I don't have telepathy, I can't see your code.

**Comment by ironman9356** [comment]
We need code to help you also please use [discussions](https://github.com/DisnakeDev/disnake/discussions) or join the [discord server](https://discord.gg/disnake) for help

**Comment by Saadelamali** [comment]
it's not about my code. Cuz it used to fine but here is my code👍:

```python

for x in os.listdir("./cogs"):
  if x.endswith(".py"):
    try:
      bot.load_extension(f"cogs.{x[:-3]}")
    except Exception:
      pass
for i in os.listdir("./antinuke"):
  if i.endswith(".py"):
    try:
      bot.load_extension(f"antinuke.{i[:-3]}")
    except Exception:
      pass
    
if __name__ == "__main__":
  bot.run(token)
```

| cogs
   | slashcommands.py
   
 ```python
 class SlashCommands(commands.Cog):
   def __init__(self, bot):
       self.bot = bot
   
   @commands.slash_command(guild_ids = [849658396675276841])
   @commands.is_owner()
   async def add(self, ctx, *, guild : int):
    premium = await cl.find_one({"_id": 849658396675276841})
    if guild in premium["list"]:
        await ctx.send("already in the premium list", ephemeral=True)
        return
    info = self.bot.get_guild(guild)    
    try:
      await cl.update_one({"_id" : 849658396675276841},{"$push" : {"list" : guild}})
      await ctx.send(f"<:zeroon1yes:866375654751338516> {info.name} added to the list.")
    except Exception as e:
      await ctx.send(f"Error: ``{e}``", ephemeral=True)
     
    
   @add.error
   async def add_error(self, ctx, error):
      if isinstance(error, commands.NotOwner):
        await ctx.send("<:zeroon1no:866375710130044968> You can't use this.", ephemeral=True)
 
        
def setup(bot):
  bot.add_cog(SlashCommands(bot))
```

**Comment by ironman9356** [comment]
What is the error in console ?

**Comment by helish88** [comment]
```
except Exception:
      pass
```
change to 
```py
except Exception as e:
      print(e)

**Comment by Saadelamali** [comment]
Thats not the issue doe

**Comment by ironman9356** [comment]
Just do it and send the issue

**Comment by Saadelamali** [comment]
it will print "cogs.slashcommands is already loaded/not loaded"

**Comment by Saadelamali** [comment]
do you even know what my issue is

> Just do it and send the issue

**Comment by ironman9356** [comment]
> do you even know what my issue is
> 
> > Just do it and send the issue

i read the issue 

your error states you are loading the same cog more than once do you know by any chance where you are doing so in your code

**Comment by Saadelamali** [comment]
> > do you even know what my issue is
> > > Just do it and send the issue
> 
> i read the issue
> 
> your error states you are loading the same cog more than once do you know by any chance where you are doing so in your code

check #help-disnake channel

**Comment by helish88** [comment]
```py
  @add.error
  async def add_error(self, ctx, error):
     if isinstance(error, commands.NotOwner):
       await ctx.send("<:zeroon1no:866375710130044968> You can't use this.", ephemeral=True)```
 ```
 add a else
 
   ```py
     @add.error
  async def add_error(self, ctx, error):
     if isinstance(error, commands.NotOwner):
       await ctx.send("<:zeroon1no:866375710130044968> You can't use this.", ephemeral=True)
     else:
         print(error)

**Comment by Saadelamali** [comment]
> ```python
>   @add.error
>   async def add_error(self, ctx, error):
>      if isinstance(error, commands.NotOwner):
>        await ctx.send("<:zeroon1no:866375710130044968> You can't use this.", ephemeral=True)```
> ```
> 
> add a else
> 
> ```python
>   @add.error
> async def add_error(self, ctx, error):
>   if isinstance(error, commands.NotOwner):
>     await ctx.send("<:zeroon1no:866375710130044968> You can't use this.", ephemeral=True)
>   else:
>       print(error)
> ```

omg thats not the issue

**Comment by helish88** [comment]
read this amazing [guide](https://guide.disnake.dev/popular-topics/errors)

**Comment by helish88** [comment]
> > ```python
> >   @add.error
> >   async def add_error(self, ctx, error):
> >      if isinstance(error, commands.NotOwner):
> >        await ctx.send("<:zeroon1no:866375710130044968> You can't use this.", ephemeral=True)```
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > add a else
> > ```python
> >   @add.error
> > async def add_error(self, ctx, error):
> >   if isinstance(error, commands.NotOwner):
> >     await ctx.send("<:zeroon1no:866375710130044968> You can't use this.", ephemeral=True)
> >   else:
> >       print(error)
> > ```
> 
> omg thats not the issue

omg, you don't see any error when you're trying to use this command,. bruh

**Comment by Saadelamali** [comment]
Anyways Thank yall!!
